worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,404 @@
+            
+        #add some extra points (useful)
+        if self.options.add_bonus_points.value == 2: #add extra points if wanted
+            already_items = len(self.itempool) + self.extra_plando_items + 1
+            self.itempool += ["Bonus Point"] * min(self.number_of_locations - already_items, 10)
+            
+        #add some story chapters (filler)
+        if self.options.add_story_chapters.value == 2: #add extra points if wanted
+            already_items = len(self.itempool) + self.extra_plando_items + 1
+            if(self.number_of_locations - already_items >= 10):
+                self.itempool += ["Story Chapter"] * 10
+                
+        #add some more extra points if there is still room
+        if self.options.add_bonus_points.value == 2:
+            already_items = len(self.itempool) + self.extra_plando_items + 1
+            self.itempool += ["Bonus Point"] * min(self.number_of_locations - already_items, 10)
+         
+        #add some encouragements filler-items if there is still room
+        already_items = len(self.itempool) + self.extra_plando_items + 1
+        self.itempool += ["Encouragement"] * min(self.number_of_locations - already_items, 5)
+
+        #add some fun facts filler-items if there is still room
+        already_items = len(self.itempool) + self.extra_plando_items + 1
+        self.itempool += ["Fun Fact"] * min(self.number_of_locations - already_items, 5)
+        
+        #finally, add some "Good RNG" and "Bad RNG" items to complete the item pool
+        #these items are filler and do not do anything.
+        
+        #probability of Good and Bad rng, based on difficulty for fun :)
+        p = 1.1 - 0.25 * self.options.game_difficulty.value
+        already_items = len(self.itempool) + self.extra_plando_items + 1
+        self.itempool += self.multiworld.random.choices(
+            ["Good RNG", "Bad RNG"], 
+            weights=[p, 1-p], 
+            k=self.number_of_locations - already_items
+        )
+
+        #we are done adding items. Now because of the last step, number of items should be number of locations
+        already_items = len(self.itempool) + self.extra_plando_items + 1
+        if already_items != self.number_of_locations:
+            raise Exception(f"[Yacht Dice] Number in self.itempool is not number of locations "
+                f"{already_items} {self.number_of_locations}.")
+        
+        #add precollected items using push_precollected. Items in self.itempool get created in create_items
+        for item in self.precollected:
+            self.multiworld.push_precollected(self.create_item(item))
+
+    def create_items(self):
+        #convert strings to actual items
+        itempool_created = [item for item in map(lambda name: self.create_item(name), self.itempool)]
+
+        #and add them to the itempool
+        for item in itempool_created:
+            self.multiworld.itempool += [item]    
```

Comment:
Ok I wasn't gonna comment on anything else yet before I'm ready to do a full review, but

*Please*. You're killing me here XD

```
def create_items(self):
    self.multiworld.itempool += [self.create_item(name) for name in self.itempool]
```

:D

Reply:
Thanks, changed!